### DISCUSSION TITLE
Elastic Issue Parsing Logs Sent from Filebeat 7.17.4

### DISCUSSION THREAD

**Discussion by clairmont32** [Security-Onion-Solutions/securityonion#8021] [Unsupported Versions] [answered]
Greetings,

After helping a friend troubleshoot why he couldnt get nginx logs sent from a separate server to the manager for a couple hours, we discovered that v7.17.4 isn't currently compatible with SO version .120. He installed filebeat from apt and configured the filebeat module and config to monitor the nginx logs correctly but after downgrading to 7.17.3 sourced from the Downloads page it worked perfectly after restoring the config backup he was using with 7.17.4.

Repro steps:
  - Install SO standalone on virtual hardware
  - Install Nginx proxy fronting a simple website
  - Install Filebeat 7.17.4 and nginx module sourced from apt
  - Configure module per documentation
  - Configure filebeat yml per SO documentation
  - Run so-allow for filebeat's portgroup per SO documentation
  - tcpdump for 5044 and nginx server source IP to observe traffic flowing in parity with filebeat logs being sent
  - Observe no logs being parsed by Elastic either in Kibana or Discover
  - Backup filebeat configs on Nginx server
  - Downgrade to 7.17.3 sourced from SO Downloads page
  - Reinstate configs from backup
  - Observe logs being parsed by Elastic in Discover or Kibana

**Comment by isaacgolding** [comment, marked as answer]
I'm pretty sure that the agents need to match their respective servers by version.  I can't quote where I saw this.  However the simple fix is to use wget or curl and pull the rpm,deb directly from the downloads page listing and install vs using the OS's repo...  Elastic both inside and outside of the SO community seems to be particularly sensitive to this issue.

**Reply by isaacgolding** [reply]
Actually here are references:
Wazuh
https://docs.securityonion.net/en/2.3/wazuh.html?highlight=agent#adding-agents

> Warning
> 
> It is important to ensure that you download the agent that matches the version of your Wazuh server. For example, if your Wazuh server is version 3.13.1, then you will want to deploy Wazuh agent version 3.13.1.


Beats
https://docs.securityonion.net/en/2.3/beats.html#beats

> Version
> When downloading a Beats agent, make sure the version number matches the version of Elastic running on your Security Onion deployment.
